Makefile

```diff
@@ -360,11 +364,11 @@ DEBUG_LDFLAGS='-extldflags "-static"'
 # $(3): The value for LDFLAGS
 define genTargetsForNativeAndDocker
 $(ISTIO_OUT)/$(1):
-	STATIC=0 GOOS=$(GOOS) GOARCH=amd64 LDFLAGS=$(3) bin/gobuild.sh $(ISTIO_OUT)/$(1) $(2)
+	STATIC=0 GOOS=$(GOOS) GOARCH=$(GOARCH) LDFLAGS=$(3) bin/gobuild.sh $(ISTIO_OUT)/$(1) $(2)
```

Comment:
I think this will break osx, we explicitly want to build linux binaries in this case?

Not 100% sure though @nmittler

Reply:
Try it also manually on osx and working fine

![image](https://user-images.githubusercontent.com/6806035/61576251-efd64b80-aad7-11e9-8c48-2ec2293f57a1.png)
![image](https://user-images.githubusercontent.com/6806035/61576264-11373780-aad8-11e9-9d25-97862f11b4d4.png)

I don't change this line https://github.com/istio/istio/blob/master/Makefile#L371, because as you said it seems that it is intended to build explicitly amd64 when output to `ISTIO_OUT_LINUX`.
I only change when it is output to `ISTIO_OUT`

Reply:
There is another command that builds explicitly amd64 using `make build-linux` and it is used in https://github.com/aliariff/istio/blob/master/tools/istio-docker.mk#L22

Reply:
@aliariff the reason it's `amd64` is for building docker targets for linux/amd64.  This will not work correctly on os.

Reply:
@aliariff I believe the `gobuild.sh` script already does this, no?

Reply:
Do you mean the arch checking? 
No there is no checking for the arch in `gobuild.sh`.
inside it is only receiving the env var `GOARCH` either it coming from export(https://github.com/aliariff/istio/blob/878328a8020088fbc5c7df8b53710f618ffe9753/Makefile#L71) or force it inline like in this case. And because of this line enforce it to `amd64`, the `gobuild.sh` use `amd64` instead of from the env

Reply:
Ok, then I think this change is reasonable. I think what @howardjohn was referring to by "break osx" was the new target that I added for docker below, which is always linix/amd64.  These changes will not affect that.

Reply:
yes that output to `ISTIO_OUT_LINUX `, I did not change that one.

Reply:
this part of the Makefile work will function properly.